packages/thyra/adapters/tpetra/src/Thyra_TpetraVectorSpace_def.hpp

```diff
@@ -114,6 +114,139 @@ TpetraVectorSpace<Scalar,LocalOrdinal,GlobalOrdinal,Node>::createMembers(int num
 }
 
 
+template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+class CopyMultiVectorViewBack {
+public:
+  CopyMultiVectorViewBack( RCP<MultiVectorBase<Scalar> > mv, const RTOpPack::SubMultiVectorView<Scalar>  &raw_mv )
```

Comment:
Now that I think about it, wouldn't it be bad to store `mv_` as an RCP?  CopyMultiVectorViewBack's destructor only kicks in whenever the RCP's reference count goes to zero.  You've avoided trouble below by creating the input RCP as nonowning (`rcpFromRef`), but in general this is unwise, I think.